ballerina/bytes_to_sse_stream_generator.bal

```diff
@@ -0,0 +1,168 @@
+        check self.byteStream.close();
+        self.isClosed = true;
+    }
+
+    private isolated function readUntilDoubleLineBreaks() returns string|error? {
+        byte[] buffer = [];
+        byte prevByte = 0;
+        byte? currentByte = ();
+        boolean foundCariageReturnWithNewLine = false;
+        while !self.isClosed {
+            currentByte = check self.getNextByte();
+            if currentByte is () {
+                return;
+            }
+            if foundCariageReturnWithNewLine && currentByte == CARRIAGE_RETURN {
+                // Lookahead for newline
+                byte? nextByte = check self.getNextByte();
+                if nextByte is () {
+                    return ();
+                }
+                if nextByte == LINE_FEED {
+                    buffer.push(currentByte);
+                    buffer.push(nextByte);
+                    return string:fromBytes(buffer);
+                }
+                // Store char in lookahead buffer if not newline
+                self.lookaheadBuffer.push(nextByte);
+            }
+            foundCariageReturnWithNewLine = false;
+            if (currentByte == LINE_FEED || currentByte == CARRIAGE_RETURN) && prevByte == currentByte {
+                buffer.push(currentByte);
+                return string:fromBytes(buffer);
+            }
+            if currentByte == LINE_FEED && prevByte == CARRIAGE_RETURN {
+                foundCariageReturnWithNewLine = true;
+            }
+            buffer.push(currentByte);
+            prevByte = currentByte;
+        }
+        return ();
+    }
+
+    # Reads next byte from the lookahead buffer if data is available, otherwise read from the byte stream
+    # + return - A `byte?` on success, `error` on failure
+    private isolated function getNextByte() returns byte|error? {
+        if self.lookaheadBuffer.length() > 0 {
+            return self.lookaheadBuffer.shift();
+        }
+        record {byte[] value;}? nextValue = check self.byteStream.next();
+        return nextValue is () ? () : nextValue.value[0];
+    }
+}
+
+isolated function parseSseEvent(string event) returns SseEvent|error {
+    string[] lines = re `\r\n|\n|\r`.split(event);
+    string? id = ();
+    string? comment = ();
+    string? data = ();
+    int? 'retry = ();
+    string? eventName = ();
+
+    foreach string line in lines {
+        if line == "" {
+            continue;
+        }
+        regexp:Groups? groups = re `(.*?):(.*)`.findGroups(line);
+        string filedName = line;
+        string fieldValue = "";
+        if groups is regexp:Groups && groups.length() == 3 {
+            regexp:Span? filedNameSpan = groups[1];
+            regexp:Span? filedValueSpan = groups[2];
+            if filedNameSpan is () || filedValueSpan is () {
+                continue;
+            }
+            filedName = filedNameSpan.substring().trim();
+            fieldValue = removeLeadingSpace(filedValueSpan.substring());
```

Comment:
Don't we need remove the trailing spaces?

Reply:
We should not remove trailing spaces here.